data/scripts.js

```diff
@@ -291,19 +291,22 @@ let BattleScripts = {
 			// 3. check for powder immunity (gen 6+ only)
 			this.hitStepPowderImmunity,
 
-			// 4. check for prankster immunity (gen 6+ only)
+			// 4. check for move-specific immunity
+			this.hitStepTryImmunityEvent,
+
+			// 5. check for prankster immunity (gen 7+ only)
 			this.hitStepPranksterImmunity,
```

Comment:
Is th4ere any reason for these not to be the same hitstep?

Reply:
poke

Reply:
If you moved the move eligibility checks inside the loop, then you could combine the powder, move-specific and prankster immunities into one function, yes.

Reply:
Like it wouldn't affect turn order for these to be the same hitstep, right?

Reply:
I don't know what the turn order is for spread move targets on cart, but there would be a minor subtle difference in that if Prankster Whimsicott used Cotton Spore against both a Dark type and a Grass type, then under the previous version the Grass type would be immune first (because Powder immunity is checked first) while under the revised version the Dark type would be immune first (because all of its immunities are checked at once).

Reply:
hrm